tests/accelerators/test_accelerator_connector.py

```diff
@@ -504,6 +504,15 @@ def test_accelerator_cpu(_):
         trainer = Trainer(accelerator="cpu", gpus=1)
 
 
+@mock.patch("torch.cuda.is_available", return_value=False)
+@pytest.mark.parametrize("devices", ["0", 0, []])
+def test_passing_zero_and_empty_list_to_devices_flag(_, devices):
+    with pytest.raises(
+        MisconfigurationException, match="GPUAccelerator can not run on your system since GPUs are not available."
```

Comment:
tbh! I wouldn't really recommend `auto` since training/inference (time) drastically varies with the accelerator. And if we recommend auto here, then users might set it and realize later that their model has been running on CPU for this whole time.

Reply:
Removing the fallback is a breaking change and this is the exception that users will see.

I suggest a hint (moreso than a recommendation) so users who break here know what they need to change to keep their previous behaviour

Reply:
but it will also pop up for the cases where let's say user doesn't have a GPU but the configuration is correct.
```py
Trainer(accelerator='gpu', devices=2)
```
recommending `auto` here is a bit off, I guess.